### DISCUSSION TITLE
Correct resizing of images

### DISCUSSION THREAD

**Discussion by thanksforthefish** [facebookresearch/detectron2#4791] [Q&A]
I train on large images (approx. 4000 x 4000 px) and downsize them in a custom mapper. However, this seems to have no effect on the training time. The training time is the same if I do not use the custom mapper.

## Instructions To Reproduce the Issue:
I use the following custom mapper:
```
def custom_mapper(dataset_dict):
    dataset_dict = copy.deepcopy(dataset_dict)  # it will be modified by code below
    image = utils.read_image(dataset_dict["file_name"], format="BGR")
    transform_list = [
        T.ResizeShortestEdge(800)
    ]
    image, transforms = T.apply_transform_gens(transform_list, image)
    dataset_dict["image"] = torch.as_tensor(image.transpose(2, 0, 1).astype("float32"))

    annos = [
        utils.transform_instance_annotations(obj, transforms, image.shape[:2])
        for obj in dataset_dict.pop("annotations")
        if obj.get("iscrowd", 0) == 0
    ]
    instances = utils.annotations_to_instances(annos, image.shape[:2])
    dataset_dict["instances"] = utils.filter_empty_instances(instances)
    return dataset_dict
```
This is my Trainer train loader:
```
    @classmethod 
    def build_train_loader(cls, cfg): 
        # mapper = None
        mapper = custom_mapper
        data_loader = build_detection_train_loader(cfg, mapper=mapper)   
        return  data_loader
```

## Expected behavior:
I expected the training time to be reduced. If I replace the images with downscaled images, the training time is significantly reduced: I use a resize factor of .25 and the time is approx. reduced by the same factor. - as expected.

Could someone please tell me how to correctly downsize the images?

## Environment:

```
----------------------  -------------------------------------------------------------------------------------------------------------------------------------
sys.platform            linux
Python                  3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]
numpy                   1.22.2
detectron2              0.6 @/caa/Homes01/holl/python_projects/segmentation-dev_seb/star_env/lib/python3.10/site-packages/detectron2
Compiler                GCC 11.2
CUDA compiler           not available
DETECTRON2_ENV_MODULE   <not set>
PyTorch                 1.12.0+cu116 @/caa/Homes01/holl/python_projects/segmentation-dev_seb/star_env/lib/python3.10/site-packages/torch
PyTorch debug build     False
GPU available           Yes
GPU 0,1,2,3,4,5,6,7     NVIDIA GeForce RTX 3090 (arch=8.6)
Driver version          520.61.05
CUDA_HOME               /opt/cuda/10-0/ - invalid!
Pillow                  9.0.1
torchvision             0.13.0+cu116 @/caa/Homes01/holl/python_projects/segmentation-dev_seb/star_env/lib/python3.10/site-packages/torchvision
torchvision arch flags  /caa/Homes01/holl/python_projects/segmentation-dev_seb/star_env/lib/python3.10/site-packages/torchvision/_C.so; cannot find cuobjdump
fvcore                  0.1.5.post20220212
iopath                  0.1.9
cv2                     4.5.5
----------------------  -------------------------------------------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.6
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.6, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF,

Testing NCCL connectivity ... this should not hang.
NCCL succeeded.
```

**Comment by github-actions** [comment]
You've chosen to report an unexpected problem or bug. Unless you already know the root cause of it, please include details about it by filling the [issue template](https://github.com/facebookresearch/detectron2/issues/new/choose).
The following information is missing: "Your Environment";

**Comment by ryouchinsa** [comment]
[RectLabel](https://rectlabel.com) is an offline image annotation tool for object detection and segmentation.
Although this is not an open source program, RectLabel can import the COCO JSON file, resize images and annotations, and export to the COCO JSON file again.

**Reply by Itisnotme9108** [reply]
Is there any app like this for andriod

**Comment by jvdgoltz** [comment]
My guess is that you also need to increase the batch size, otherwise there will not be much gain in reducing the image resolution. If you train on half the image resolution, you can approximately quadruple the batch size (cause GPU memory scales roughly quadratically with the resolution), that should give you a good increase in speed.

Keep in mind that you also have to change the learning rate if you change the batch size
